packages/remix-dev/vite/plugin.ts

```diff
@@ -40,6 +40,12 @@ import { resolveFileUrl } from "./resolve-file-url";
 import { removeExports } from "./remove-exports";
 import { importViteEsmSync, preloadViteEsm } from "./import-vite-esm-sync";
 
+function combineURLs(baseURL: string, relativeURL: string) {
+  return relativeURL
+      ? baseURL.replace(/\/+$/, '') + '/' + relativeURL.replace(/^\/+/, '')
+      : baseURL;
```

Comment:
Could you add a comment with some examples of why this approach is needed for future context if folks are in this file?

Reply:
I've added some unit and integration tests to capture this instead.